pkg/controllers/leasegarbagecollection/controller.go

```diff
@@ -0,0 +1,71 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package leasegarbagecollection
+
+import (
+	"context"
+
+	v1 "k8s.io/api/coordination/v1"
+	"knative.dev/pkg/logging"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	corecontroller "github.com/aws/karpenter-core/pkg/operator/controller"
+)
+
+// Controller for the resource
+type Controller struct {
+	kubeClient client.Client
+}
+
+// NewController is a constructor
+func NewController(kubeClient client.Client) corecontroller.Controller {
+	return corecontroller.Typed[*v1.Lease](kubeClient, &Controller{
+		kubeClient: kubeClient,
+	})
+}
+
+func (c *Controller) Name() string {
+	return "lease.garbagecollection"
+}
+
+// Reconcile the resource
+func (c *Controller) Reconcile(ctx context.Context, l *v1.Lease) (reconcile.Result, error) {
+	if l.OwnerReferences != nil || l.Namespace != "kube-node-lease" {
+		return reconcile.Result{}, nil
+	}
+	err := c.kubeClient.Delete(ctx, l)
+	if err == nil {
+		logging.FromContext(ctx).Infof("found and delete leaked lease: %s", l.Name)
```

Comment:
```suggestion
		logging.FromContext(ctx).Debug("found and delete leaked lease")
```

I think `Info` may be too high here. Also, the lease name is already injected in the logger by default. Have we tested this and ensured that the log looks right when this happens?